src/components/ui/Paginator/Paginator.tsx

```diff
@@ -0,0 +1,213 @@
+        pageRef.current.removeChild(pageRef.current.firstChild)
+      }
+
+      // if there is only one element, and it has a scrollTo attribute, add that element and scroll it
+      if (
+        pages[currentPage].length == 1 &&
+        (pages[currentPage][0] as HTMLDivElement).getAttribute('scrollTo') !== null
+      ) {
+        const pageDiv = pages[currentPage][0] as HTMLDivElement
+        const scrollTo = pageDiv.getAttribute('scrollTo')
+
+        pageRef.current.appendChild(pageDiv)
+
+        pageRef.current.scrollTo({
+          left: orientation === 'horizontal' ? Number(scrollTo) : 0,
+          top: orientation === 'vertical' ? Number(scrollTo) : 0,
+          behavior: 'smooth',
+        })
+      } else {
+        // otherwise, just add all the current page elements
+        pages[currentPage].forEach((el: Node) => {
+          if (pageRef.current !== null) pageRef.current.appendChild(el)
+        })
+      }
+    }
+  }, [pages, currentPage, orientation])
+
+  return (
+    <div
+      ref={wrapperRef}
+      className={classnames('w-full h-full flex justify-between', {
+        'flex-row': selectorLocation.startsWith('right'),
+        'flex-col': selectorLocation.startsWith('bottom'),
+        'flex-row-reverse': selectorLocation.startsWith('left'),
+        'flex-col-reverse': selectorLocation.startsWith('top'),
+      })}
+    >
+      {(pages.length == 0 || pages.length == 1) && (
+        <div
+          className={classnames('flex', {
+            'flex-row': orientation === 'horizontal',
+            'flex-col': orientation === 'vertical',
+          })}
+        >
+          {childrenArray.map((child, i) => (
+            <div
+              key={i}
+              ref={(el) => {
+                if (el !== null) childrenRef.current[i] = el as HTMLDivElement
+              }}
+            >
+              {child}
+            </div>
+          ))}
+        </div>
+      )}
+      {pages.length > 1 && (
+        <div
+          ref={pageRef}
+          className={classnames('overflow-hidden flex', {
+            'flex-row': orientation === 'horizontal',
+            'flex-col': orientation === 'vertical',
+            'h-[calc(100%-54px)] w-full': selectorLocation.startsWith('bottom') || selectorLocation.startsWith('top'),
+            'w-[calc(100%-54px)] h-full': selectorLocation.startsWith('right') || selectorLocation.startsWith('left'),
+          })}
+        ></div>
+      )}
+      {pages.length > 1 && (
+        <PageSelector
+          onClickLeft={() => {
+            setCurrentPage(currentPage - 1)
+          }}
+          onClickRight={() => {
+            setCurrentPage(currentPage + 1)
+          }}
+          currentPage={currentPage}
+          maxPages={pages.length}
+          selectorLocation={selectorLocation}
+        ></PageSelector>
+      )}
+    </div>
+  )
+}
+
+interface Props {
+  children: JSX.Element[] | JSX.Element
+  orientation: 'vertical' | 'horizontal'
```

Comment:
Not optional, but a default is provided in the component, either get rid of the default (line 6) or make this optional (`orientation?:`). Same goes for `selectorLocation`

Reply:
you're right, should be optional, thanks